### PR TITLE
Frame computation calculation fixing

### PR DESCRIPTION
**Pull request by Yash-10**
In this pull request, I have tried to rectify the mistake done by me in #1190 .

**Details**

- The `core` module uses the `np.deg2rad` function to address the needed conversion. It is done since `numba` loves `numpy` and the use of `astropy.Quantity` objects in `core`, I think, is discouraged.
- The `epoch` argument in `rot_elements_at_epoch` is made optional since the docstring earlier signified it to be optional, but the code implicitly assumed it to be always passed by the user.
- A test has been added to make sure the implementation works. (This might contribute a small amount to #1236). The expected values for the test were taken from the output of using `rot_elements_at_epoch` on fixed frames of bodies using the source code before the commit of #1190.

@jorgepiloto I would love to get your thoughts on this!

Thanks!

---
If I run the added test but keeping the code in `core` as per the commits in #1190, I get error:

```bash
FAILED tests/test_frames.py::test_fixed_frame_calculation_gives_expected_result[body1-MercuryFixed-radecW1] - AssertionError: 
FAILED tests/test_frames.py::test_fixed_frame_calculation_gives_expected_result[body3-MarsFixed-radecW3] - AssertionError: 
FAILED tests/test_frames.py::test_fixed_frame_calculation_gives_expected_result[body4-JupiterFixed-radecW4] - AssertionError: 
FAILED tests/test_frames.py::test_fixed_frame_calculation_gives_expected_result[body7-NeptuneFixed-radecW7] - AssertionError: 
FAILED tests/test_frames.py::test_fixed_frame_calculation_gives_expected_result[body8-MoonFixed-radecW8] - AssertionError: 
```
Hence, this might be an indication that this pull request would probably resolve the issue.

### CODE REVIEW

**Review by astrojuanlu — APPROVED** [review]
Looks good to me, but waiting for @jorgepiloto to give the final review!

File: src/poliastro/core/fixed.py
Comment (jorgepiloto) [inline src/poliastro/core/fixed.py:63]: Looks good to me too! However, I would like to suggest one thing and know your opinion about it: we are calling several times the `np.deg2rad` function, what about implementing the following at the very beginning of the file:

```python
DEG2RAD = np.deg2rad(1)
```
so we can do:

```python
M1 = number_in_degrees * DEG2RAD
```

What do you think?

Comment (astrojuanlu) [reply src/poliastro/core/fixed.py:63]: All these functions are jitted, so I don't expect a big difference. However, I might be wrong :)

Comment (Yash-10) [reply src/poliastro/core/fixed.py:63]: The execution times seem to be almost the same with the approach of using `numpy` multiple times being marginally faster (`~1e-2 s`). From a short analysis, this marginal increase might be because

```py
DEG2RAD = np.deg2rad(1)
```
would not be inside any jitted function and hence introduces some more execution time, although using just `DEG2RAD` inside the jitted functions is slightly faster than this PR's implementation.

As @astrojuanlu said, the differences are not that big, so I wonder what option could we choose here?

Comment (jorgepiloto) [reply src/poliastro/core/fixed.py:63]: Yeah, let us use the original `np.deg2rad` implementation. Thanks for the comments 🚀

**Review by jorgepiloto — APPROVED** [review]
Merging this, thanks for addressing it @Yash-10 👍🏽

**Comment by jorgepiloto** [comment]
Now, paying attention to the latest validation actions run, see https://github.com/poliastro/validation/actions/runs/895277143 🤞🏽

**Comment by jorgepiloto** [comment]
All checks passing 🎉

**Pull request merged**
